### ISSUE TITLE
can I accomplish any Parallelization Schemes with 4 GPUs using Caffe (NVIDIA's fork)

### ISSUE THREAD

**Issue by MagnusBai** [NVIDIA/caffe#14]
Nowadays, I am looking for an available parallelization version of Caffe, in order to speed up training.

I tried to use Caffe (NVIDIA's fork) by the following command: 

build/tools/caffe train -gpus all -solver ......

VOC2012's images were used as train/val data, and the (softmax) loss was logged iteration by iteration.
In order to compare the performance( speed of convergence ), I run the training with 4 GPUs and 1 GPU and plot the loss/iteration on the graph:

![gpus_plot](https://cloud.githubusercontent.com/assets/12607834/9060449/8c5c73aa-3ae6-11e5-84bd-bee4e1b2157f.jpg)

the red line is the loss with 1 GPU and the rest are 4 GPUs'.

I want to know why 4 GPUs didn't help with the speed of convergence in my task.

**Comment by slayton58** [comment]
Speed of convergence is not a measure of performance in the sense you're using it, and is  (ideally) completely independent of the use of multi-GPU. Essentially "speed of convergence" is showing how good your network is at converging to a solution, NOT how fast (wall-clock time) you actually reached that solution.

Your plot shows that the 4 GPU run is converging to the solution as quickly as single GPU runs -- this is good to see.

Where you should be looking for an improvement in multi-GPU runs, is how long it took in wall clock time to get to the same solution. Look at how long it took you to perform 100k iterations on a single GPU and compare that to how long it took to perform 100k iterations on 4 GPUs - the 4 GPU run should have taken less time.
